src/import-utils.ts

```diff
@@ -217,6 +237,37 @@ export function rewriteAstBaseUrl(
 }
 
 /**
+ * Walk through inline scripts of an import document.
+ * For each script create identity source maps unless one already exists.
+ *
+ * The generated script mapping detail is the relative location within
+ * the script tag. Later this will be updated to account for the
+ * line offset within the final bundle.
+ */
+export async function addSourcemapsToInlineScripts(
+    document: Document, ast: ASTNode, oldBaseUrl: UrlString) {
+  const inlineScripts = dom5.queryAll(ast, matchers.inlineJavascript);
+  const promises = inlineScripts.map(scriptAst => {
+    let content = dom5.getTextContent(scriptAst);
+    const scriptLocation =
+        (scriptAst.__location as parse5.ElementLocationInfo).startTag;
```

Comment:
There's a number of subtle issues with parse5's __locations that we handle in the analyzer. Recommend this instead:

```typescript
const parsedHtmlDocument = document.parsedDocument as ParsedHtmlDocument;
const sourceRange = parsedHtmlDocument.sourceRangeForStartTag(scriptAst)!;
```

That gets you a zero-indexed source range from the start to the end of the element's start tag. The biggest thing I notice that's relevant here is that it will handle newlines in the attribute section of a start tag correctly.